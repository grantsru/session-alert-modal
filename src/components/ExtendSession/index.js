import React from "react";
import PropTypes from "prop-types";

import Button from "@material-ui/core/Button";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import { withStyles } from "@material-ui/core/styles";

import Countdown from "../Countdown";

const styles = {
  warningText: {
    margin: "10px 0"
  }
};

const ExtendSession = ({ classes, click, extend, logout, timeUntilExpired, warningText }) => (
  <React.Fragment>
    <Typography className={classes.warningText} variant={"subheading"}>
      {warningText}
    </Typography>
    <Countdown time={timeUntilExpired}  />
    <Grid container justify={"center"} spacing={16}>
      <Grid item>
      <Button
        color={"primary"}
        variant={"contained"}
        onClick={() => click(extend)}>
        Extend Session
      </Button>
      </Grid>
      {
        !!logout &&
        <Grid item>
          <Button
            color={"default"}
            variant={"contained"}
            onClick={() => click(logout)}>
            Logout
          </Button>
        </Grid>
      }
    </Grid>
  </React.Fragment>
);

ExtendSession.propTypes = {
  click: PropTypes.func.isRequired,
  extend: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  timeUntilExpired: PropTypes.number.isRequired,
  warningText: PropTypes.string.isRequired
};

export default withStyles(styles)(ExtendSession);
