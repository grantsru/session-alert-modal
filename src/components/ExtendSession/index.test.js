import React from "react";
import ExtendSession from "./index";
import { create } from "react-test-renderer";

const logout = () => null;
const extend = () => null;
const handleButtonClick = () => null;
const warningText = "foo";
const timeUntilExpired = 10;

describe("ExtendSession", () => {
  test("it matches the snapshot", () => {
    const component = create(
      <ExtendSession
        extend={extend}
        logout={logout}
        warningText={warningText}
        timeUntilExpired={timeUntilExpired}
        click={handleButtonClick}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
