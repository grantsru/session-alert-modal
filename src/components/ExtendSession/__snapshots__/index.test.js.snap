// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtendSession it matches the snapshot 1`] = `
Array [
  <h3
    className="MuiTypography-root-2 MuiTypography-subheading-9 ExtendSession-warningText-1"
  >
    foo
  </h3>,
  <h3
    className="MuiTypography-root-2 MuiTypography-subheading-9 MuiTypography-alignCenter-25"
    style={
      Object {
        "margin": "15px 0 20px 0",
      }
    }
  >
    <strong>
      Time Remaining:
    </strong>
     
    10 seconds
  </h3>,
  <div
    className="MuiGrid-container-38 MuiGrid-spacing-xs-16-61 MuiGrid-justify-xs-center-55"
  >
    <div
      className="MuiGrid-item-39"
    >
      <button
        className="MuiButtonBase-root-161 MuiButton-root-135 MuiButton-contained-146 MuiButton-containedPrimary-147 MuiButton-raised-149 MuiButton-raisedPrimary-150"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-136"
        >
          Extend Session
        </span>
        <span
          className="MuiTouchRipple-root-164"
        />
      </button>
    </div>
    <div
      className="MuiGrid-item-39"
    >
      <button
        className="MuiButtonBase-root-161 MuiButton-root-135 MuiButton-contained-146 MuiButton-raised-149"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-136"
        >
          Logout
        </span>
        <span
          className="MuiTouchRipple-root-164"
        />
      </button>
    </div>
  </div>,
]
`;
